pkg/tests/ossm/operator_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ossm
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/maistra/maistra-test-tool/pkg/util"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+var workername string
+
+// TestOperator tests scenario to cover all the test cases related to the OSSM operators
+func TestOperator(t *testing.T) {
+	NewTest(t).Id("T40").Groups(Full).Run(func(t TestHelper) {
+		t.Log("Test related to OSSM Operators and Istio resources")
+		if Smcp.Rosa {
+			t.Skip("Skipping test on ROSA") // Now in Rosa this test need to be skipped due to lack of permissions in ROSA cluster
+		}
+		t.Cleanup(func() {
+			oc.Label(t, "", "node", workername, "node-role.kubernetes.io/infra-")
+			oc.Label(t, "", "node", workername, "node-role.kubernetes.io-")
+		})
+
+		t.LogStep("Setup: Get a worker node from the cluster that does not have the istio operator installed, label it as infra")
+		workername = pickWorkerNode(t)
+		oc.Label(t, "", "node", workername, "node-role.kubernetes.io/infra=")
+		oc.Label(t, "", "node", workername, "ode-role.kubernetes.io=infra")
+
+		// Reference: https://issues.redhat.com/browse/OSSM-2342
+		t.NewSubTest("Run on Infra Nodes").Run(func(t TestHelper) {
+			t.Log("Testing: Run OSSM Operator on infra nodes")
+			t.Cleanup(func() {
+				// Untaint the infra node and remove modification in subscription
+				shell.Execute(t,
+					`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule- node-role.kubernetes.io/infra=reserved:NoExecute-`)
+				shell.Execute(t,
+					`oc patch subscription servicemeshoperator -n openshift-operators --type json -p='[{"op": "remove", "path": "/spec/config/tolerations"}]'`)
+				oc.WaitSMCPReady(t, meshNamespace, smcpName) // Wait for the SMCP to be ready before move to next subtest case (Because untaint)
+			})
+			t.LogStep("Taint node and edit the subscription to add the infra node to the node selector")
+			shell.Execute(t,
```

Comment:
Why don't you taint in the test setup? That's where this belongs (immediately after you label the node)